src/main/java/io/vertx/ext/mail/MailMessage.java

```diff
@@ -0,0 +1,250 @@
+package io.vertx.ext.mail;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+@DataObject
+public class MailMessage {
+
+  private String bounceAddress;
+  private String from;
+  private List<String> to;
+  private List<String> cc;
+  private List<String> bcc;
+  private String subject;
+  private String text;
+  private String html;
+  private List<MailAttachment> attachment;
+
+  public MailMessage() {
+  }
+
+  public MailMessage(MailMessage other) {
+    Objects.requireNonNull(other);
+    this.bounceAddress = other.bounceAddress;
+    this.from = other.from;
+    this.to = other.to;
+    this.cc = other.cc;
+    this.bcc = other.bcc;
+    this.subject = other.subject;
+    this.text = other.text;
+    this.html = other.html;
+    if (other.attachment != null) {
+      List<MailAttachment> newList = new ArrayList<MailAttachment>();
```

Comment:
minor : create an array list with the size since you know the other.attachment size